### PR TITLE
abstract firestore to repository layer

### PR DESCRIPTION
**Pull request by Mr1upMachine**
This would move all firebase specific logic out of the commands and into several db interaction functions. This is inspired by React hook syntax for getting global state values & by spring's jpa syntax

### CODE REVIEW

File: src/services/global-state.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import type { CronJob } from 'cron';
+
+interface GlobalState {
+  activityCronJob?: CronJob;
+  firestore?: Firestore;
+  randomVoiceChannelNameCronMap: Map<string, CronJob>;
+}
+
+// TODO find a better way to do this
+export const GLOBAL_STATE: GlobalState = {
```
Comment (Mr1upMachine) [inline src/services/global-state.ts:11]: I am probably going to create hook style reducer functions for these too, but not 100% sure.

**Review by cstrahl — APPROVED** [review]
derge

File: src/commands/edit-profile/index.ts [hunk 2]
```diff
@@ -54,20 +51,17 @@ export class EditProfileCommand extends Command {
           );
         }
 
-        await userFirestoreRef.set({ about }, { merge: true });
+        await setDBGuildMember({ about });
```
Comment (Mr1upMachine) [inline src/commands/edit-profile/index.ts:54]: TODO update this to collect all fields to update and only make a single `setDBGuildMember` call

File: src/db/guilds-repository.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { BaseRepository } from './base-repository.js';
+import type { DBGuild } from './types.js';
+import { useFirestore } from './use-firestore.js';
+
+export class GuildsRepository extends BaseRepository<DBGuild> {
+  constructor() {
+    super(useFirestore(), 'guilds');
+  }
+
+  async findWhereRandomVoiceChannelNamesIs(
```
Comment (cstrahl) [inline src/db/guilds-repository.ts:10]: findByRandomVoiceChannelName

**Review by cstrahl — APPROVED** [review]

**Review by IAmBitfrost — APPROVED** [review]

**Pull request merged**